### DISCUSSION TITLE
Import Midi ???

### DISCUSSION THREAD

**Discussion by dtrey** [hydrogen-music/hydrogen#1277] [Q&A]
Hello, does Hydrogen allow for pre-recorded midi files to be imported. I have many midi files which need to be bounced to a wav file for band mates to practice with.
Thank you

**Comment by mauser** [comment]
Hi! 

no, Hydrogen is not able to import midi files. It can only export existing Hydrogen songs to midi.

**Comment by theGreatWhiteShark** [comment]
The reason why its not supported is that there is no way to know which of the different channels in your MIDI file should map to which instrument of which Hydrogen drumkit.

But you can nevertheless make it happen:
1. Load the MIDI file into an arbitrary program supporting MIDI output
2. Connect the MIDI input of Hydrogen with the MIDI output of the program in 1.
3. Adjust the `Channel` and `Note` values in the "General" tab of the Instrument Editor to match the channels of your MIDI file
4. Capture the audio output of Hydrogen with another program (using the *Export Song* option does not work for this "live" usage)

**Comment by NikicaJurcevic** [comment]
Hi. A couple of points:

What is the function?
Is there something that would be standard mapping?

For me answer to both is Drums From Hell. It's a legendary library with studio quality close and room drum samples and MIDI tracks by real drummers.  I used to integrate that in a complicated environment (Cubase/Battery). The simplicity of Hydrogen is appealing and I think that would make me more creative. I would love to have those DFH midi files split and saved as h2patterns.

**Reply by theGreatWhiteShark** [reply]
Hey @NikicaJurcevic,

There has actually been quite some work since this discussion was started. In version 2 of Hydrogen a MIDI import for individual patterns would be possible and most probably a good idea too. I'll add it to the long list of features. Not sure if it can be realized in time for 2.0 though.

**Comment by bassklampfe** [comment]
Please see #2126  for a solution.
